pkg/publicapi/endpoint/agent/endpoint.go

```diff
@@ -138,3 +143,30 @@ func (e *Endpoint) config(c echo.Context) error {
 		Config: cfg,
 	})
 }
+
+// license godoc
+//
+//	@ID			agent/license
+//	@Summary	Returns the details of the current configured orchestrator license.
+//	@Tags		Ops
+//	@Produce	json
+//	@Success	200	{object}	apimodels.GetAgentLicenseResponse
+//	@Failure	500	{object}	string
+//	@Router		/api/v1/agent/license [get]
+func (e *Endpoint) license(c echo.Context) error {
+	if e.licenseManager == nil {
```

Comment:
Let's add null validation in the constructor and fail early

Reply:
@wdbaruni since `licenseManager` is a pointer, I think it is better to validate for null each time to avoid panics if another entity manipulated it somehow.

Reply:
That is unnecessary complexity since the variable is immutable. We still need to validate at the constructor level even if you feel we need to validate here as well

Reply:
Done. Null validation in the constructor now.